idaes/apps/grid_integration/examples/thermal_generator_storage.py

```diff
@@ -0,0 +1,1069 @@
+import pyomo.environ as pyo
+import pandas as pd
+from utils import df_col_to_dict, get_data_given
+from collections import deque
+
+def _check_number_of_None(ls):
+
+    cnt = 0
+
+    for item in ls:
+        if item is None:
+            cnt += 1
+
+    return cnt
+
+def assemble_model_data(generator_names, \
```

Comment:
What do you think about, after generalizing, making one of the examples this thermal generator with optional storage? I am okay with RTS-GMLC data living in that example.